rosbag2_cpp/src/rosbag2_cpp/rmw_implemented_serialization_format_converter.cpp

```
@@ -122,6 +122,14 @@ void RMWImplementedConverter::deserialize(
   const rosidl_message_type_support_t * type_support,
   std::shared_ptr<rosbag2_cpp::rosbag2_introspection_message_t> introspection_message)
 {
+  if (!serialized_message || !serialized_message->serialized_data) {
+    ROSBAG2_CPP_LOG_ERROR("Serialized message is null.");
+    return;
+  }
+  if (!introspection_message || !introspection_message->message) {
+    ROSBAG2_CPP_LOG_ERROR("Introspection message is null.");
+    return;
+  }
```

I feel like throwing an exception instead of logging an error and not doing anything might be better and lead to fewer weird side effects.

---

Well, if input is null, there is nothing to do, and it is okay just to log a error or warning. It could be a valid case for an empty message.
However, if memory for output message hasn't been allocated and we can't write the result of the serialization/deserialization due to that - that is kind of situation which shouldn't happened and we should throw an exception.